[3928] Reactify History dialog

aachavarria opened the pull request:
creating historyDialog to work with Redux
https://github.com/craftercms/craftercms/issues/3928

aachavarria committed:
creating historyDialog to work with Redux

aachavarria committed:
changes on HistoryDialog to use redux state

aachavarria committed:
- Select a revision to compare" state should not show item menu
- Closing the dialog should clean the dialog state
- Make the dialog's body the scrollable piece
- Should not show pagination/footer
- Add back to history list button

aachavarria committed:
- Add back to history list button
- Add tooltips to dialog header actions
- Close should become a FastRewind icon
TODO: issue on backToHistoryList btn

aachavarria committed:
- add revertContent (WIP)

aachavarria committed:
changes on the HistoryDialog resource,
adding revertContent;
revertContentActions and Epic;
pending epic changes to call fetch

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3928

rart committed:
Code format

rart committed:
Code format

rart commented on ui/app/src/state/epics/dialogs.ts:
```diff
@@ -23,10 +23,25 @@ import { closeNewContentDialog } from '../reducers/dialogs/newContent';
-  name = name.substring(0, name.indexOf("_"));
+  let name = type.substring(type.indexOf('_') + 1);
+  name = name.substring(0, name.indexOf('_'));
   return camelize(name.toLowerCase());
```
This isn't working for compound names (e.g. newContent). It's fixed now, pull.

rart commented on ui/app/src/state/reducers/dialogs/history.ts:
```diff
@@ -17,25 +17,105 @@
+  rowsPerPage: 10,
+  page: 0,
+  path: null,
+  compareAB: {
```
compareAB => compare

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3928

aachavarria committed:
Merge remote-tracking branch 'aaron/feature/3928' into feature/3928

aachavarria committed:
Renaming compareAB to compare;
revertContentComplete epic;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3928

 Conflicts:
	ui/app/src/state/epics/dialogs.ts

aachavarria committed:
Renaming compareAB to compare;

aachavarria committed:
adding revertToPrevious, compareToPrevious, sort

aachavarria committed:
Refactoring HistoryDialog on ViewVersionDialog and CompareVersionsDialog(WIP)
HistoryDialog.tsx: moving VersionList, creating Pagination Wrapper;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3928

 Conflicts:
	ui/app/src/components/DialogHeader.tsx

aachavarria committed:
adding viewVersionDialog(WIP);
adding getContentVersion service;
adding newActions to Hide and Reveal historyDialog;

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3928

 Conflicts:
	ui/app/src/services/content.ts

aachavarria committed:
Adding ContentTypes to viewVersionDialog.tsx;
DialogHeader.tsx: adding DialogHeaderStateAction;

aachavarria committed:
CompareVersionsDialog wip

aachavarria committed:
adding CompareVersionsDialog;
Adding Versions to reducer; and redux store;
Pending work on versions pagination(WIP);

aachavarria committed:
changes on versionList resource WIP

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3928

 Conflicts:
	ui/app/src/components/PagesWidget.tsx
	ui/app/src/models/GlobalState.ts
	ui/app/src/modules/Content/History/HistoryDialog.tsx
	ui/app/src/services/content.ts

aachavarria committed:
Merge branch 'develop' of https://github.com/craftercms/studio-ui into feature/3928

aachavarria committed:
MergeChanges

aachavarria committed:
adding new historyDispatch to Widget.tsx

aachavarria committed:
Fixing name of the import

aachavarria committed:
changes on DialogHeader to support stringMap icons;
Widget.tsx change showHistoryDialog route;
GlobalDialogManager.tsx: removing unnecessary props, adding selected to compareDialog;
ComponentMenu.tsx: chaning route of showHistoryDialog;
GlobalState: adding VersionsStateProps;
adding CompareVersionsDialog;
HistoryDialog.tsx: changes on recourse;
VersionsList.tsx: changes on recourse, adding support to selected;
adding actions/dialogs.ts;
adding compare versions Epics;
changes on compareVersion, history, viewVersions reducers, versions ;

aachavarria committed:
adding RevertToThisVersions, compareToCurrent

aachavarria committed:
adding compareBoth;
Pending Compare/Revert Previous;

rart commented on ui/app/src/components/Dialogs/DialogHeader.tsx:
```diff
@@ -77,35 +86,14 @@ const translations = defineMessages({
-  SecondaryTypographyComponent extends React.ElementType = 'p'
-> = PropsWithChildren<{
+export type DialogTitleProps<PrimaryTypographyComponent extends React.ElementType = 'h2',
+  SecondaryTypographyComponent extends React.ElementType = 'p'> = PropsWithChildren<{
```
Format with prettier

rart commented on ui/app/src/components/Dialogs/DialogHeader.tsx:
```diff
@@ -77,35 +86,14 @@ const translations = defineMessages({
+  titleTypographyProps?: TypographyProps<PrimaryTypographyComponent,
+    { component?: PrimaryTypographyComponent }>;
+  subtitleTypographyProps?: TypographyProps<SecondaryTypographyComponent,
+    { component?: SecondaryTypographyComponent }>;
```
Format with prettier

aachavarria committed:
DialogHeader.tsx prettier format;

aachavarria committed:
GlobalDialogManager: adding selected versions;
WIP previous;
Changes on compareVersions to support deselect;
Changes on VersionsResource to support PageChanged without Api;

aachavarria committed:
GlobalDialogManager.tsx, dialogs.ts, versions.ts, history.ts: moving the revert logic to versions.ts;
versions.ts: adding previous version number to use it on reverToPrevious;
HistoryDialog.tsx: adding compareToPrevious, reverToPrevious;

aachavarria committed:
adding validation when there are only 1 version;

rart commented on ui/app/src/state/epics/dialogs.ts:
```diff
@@ -38,20 +48,37 @@ function getDialogState(type: string, state: GlobalState): any {
+    ),
+  // endregion
+  // region View Version Dialog
+  // region View Version Dialog
```
Duplicate

rart replied:
Send update on the next PR.

aachavarria committed:
removing Duplicate;
